test/integration/acoustic_comms.cc

```diff
@@ -0,0 +1,188 @@
+
+  // Publish 10 messages.
+  gz::msgs::StringMsg payload;
+  unsigned int pubCount = 10u;
+  for (unsigned int i = 0u; i < pubCount; ++i)
+  {
+    // Prepare the payload.
+    payload.set_data("hello world " + std::to_string(i));
+    std::string serializedData;
+    EXPECT_TRUE(payload.SerializeToString(&serializedData))
+        << payload.DebugString();
+    msg.set_data(serializedData);
+    EXPECT_TRUE(pub.Publish(msg));
+    server.Run(true, 100, false);
+  }
+
+  // Verify subscriber received all msgs.
+  int sleep = 0;
+  bool done = false;
+  while (!done && sleep++ < 10)
+  {
+    std::lock_guard<std::mutex> lock(mutex);
+    done = msgCounter == pubCount;
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  }
+  EXPECT_EQ(pubCount, msgCounter);
+}
+
+/////////////////////////////////////////////////
+TEST_F(AcousticCommsTest,
+    GZ_UTILS_TEST_DISABLED_ON_WIN32(AcousticCommsOutOfRange))
+{
+  // Start server
+  ServerConfig serverConfig;
+  const auto sdfFile =
+    gz::common::joinPaths(std::string(PROJECT_SOURCE_PATH),
+      "examples", "worlds", "acoustic_comms.sdf");
+  serverConfig.SetSdfFile(sdfFile);
+
+  Server server(serverConfig);
+  EXPECT_FALSE(server.Running());
+  EXPECT_FALSE(*server.Running(0));
+
+  // Run server
+  size_t iters = 1000;
+  server.Run(true, iters, false);
+
+  unsigned int msgCounter = 0u;
+  std::mutex mutex;
+  auto cb = [&](const msgs::Dataframe &_msg) -> void
+  {
+    // Verify msg content
+    std::lock_guard<std::mutex> lock(mutex);
+    msgCounter++;
+  };
+
+  // Create subscriber.
+  gz::transport::Node node;
+  std::string addr  = "addr3";
+  std::string subscriptionTopic = "addr3/rx";
+
+  // Subscribe to a topic by registering a callback.
+  auto cbFunc = std::function<void(const msgs::Dataframe &)>(cb);
+  EXPECT_TRUE(node.Subscribe(subscriptionTopic, cbFunc))
+      << "Error subscribing to topic [" << subscriptionTopic << "]";
+
+  // Create publisher.
+  std::string publicationTopic = "/broker/msgs";
+  auto pub = node.Advertise<gz::msgs::Dataframe>(publicationTopic);
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  // Prepare the message.
+  gz::msgs::Dataframe msg;
+  msg.set_src_address("addr4");
+  msg.set_dst_address(addr);
+
+  // Publish 10 messages.
+  gz::msgs::StringMsg payload;
+  unsigned int pubCount = 10u;
+  for (unsigned int i = 0u; i < pubCount; ++i)
+  {
+    // Prepare the payload.
+    payload.set_data("hello world " + std::to_string(i));
+    std::string serializedData;
+    EXPECT_TRUE(payload.SerializeToString(&serializedData))
+        << payload.DebugString();
+    msg.set_data(serializedData);
+    EXPECT_TRUE(pub.Publish(msg));
+    server.Run(true, 100, false);
+  }
+
+  // Verify subscriber received no msgs.
+  int sleep = 0;
+  while (sleep++ < 5)
+  {
+    std::lock_guard<std::mutex> lock(mutex);
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  }
+  EXPECT_EQ(0, msgCounter);
+}
```

Comment:
@adityapande-1995 meta: it may be worth adding a test with a moving sender and another test with a moving receiver.

Reply:
Done : https://github.com/gazebosim/gz-sim/pull/1608/commits/920506df253f0ea251ae5b9a6e7d5bccc0522594